internal/pkg/externalplugins/blunderbuss/blunderbuss_test.go

```diff
@@ -186,63 +275,157 @@ func TestHandleIssueCommentEvent(t *testing.T) {
 
 func TestHandlePullRequest(t *testing.T) {
 	var testcases = []struct {
-		name              string
-		action            github.PullRequestEventAction
-		body              string
-		maxReviewersCount int
-		excludeReviewers  []string
+		name   string
+		action github.PullRequestEventAction
+		body   string
+		state  string
+		// labels specifies the labels the PR already owned.
+		labels []string
+		// label specifies the label related to labeled and unlabeled events.
+		label string
+		// Whether to simulate other plugins in the sleep function to add tags to the current PR
+		mockOtherPluginAddSigLabel bool
```

Comment:
I find this a bit hard to understand, can we not simulate this event?

I think it unnecessary.

Reply:
However, the existence of SIG label determines that the logic behind open event can continue to execute. If we want to test it, we need to simulate it.

Reply:
Can we just add it? Do not combine with sleep.

Reply:
It's a bit hard to understand.

Reply:
In this case, we need to find a suitable time to add the sig label. If it is added before calling the HandlePullRequestEvent function, it will make people mistakenly think that the PR is labeled in the open event.

Reply:
Maybe mock the containSigLabel function and through an option to directly control its return results?

Reply:
OK, got it.

`mockOtherPluginAddSigLabel` to `mockAddSigLabel` maybe better?